### PR TITLE
Create a github issue when someone reports a bug

### PR DESCRIPTION
**Pull request by richvdh**
Just a placeholder for now - making it more useful will be a separate PR.

Suggest reviewing the commits individually

### CODE REVIEW

File: src/github.com/matrix-org/rageshake/submit.go [hunk 1]
```diff
@@ -51,6 +57,9 @@ func (s *submitServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		respond(405, w)
 		return
 	}
+
+	ctx := req.Context()
```
Comment (Kegsay) [inline src/github.com/matrix-org/rageshake/submit.go]: Why is this sooooo faaaar aaawaaay from its usage point on `:81`?

File: src/github.com/matrix-org/rageshake/submit.go [hunk 2]
```diff
@@ -96,6 +106,27 @@ func saveReport(p payload) error {
...
 	}
+
+	if s.ghClient == nil {
+		// we're done here
+		return nil
+	}
+
+	// submit a github issue
+	owner := "richvdh"
+	repo := "test"
```
Comment (Kegsay) [inline src/github.com/matrix-org/rageshake/submit.go]: This obviously needs to be yanked out to an env var.

File: src/github.com/matrix-org/rageshake/submit.go [hunk 3]
```diff
@@ -96,6 +106,27 @@ func saveReport(p payload) error {
...
+	title := "Automated bug report"
+	issueReq := github.IssueRequest{
+		Title: &title,
+	}
```
Comment (Kegsay) [inline src/github.com/matrix-org/rageshake/submit.go]: No description?

File: README.md [hunk 4]
```diff
@@ -2,16 +2,26 @@
...
 ```
 
+where:
+
+ * `token` is a GitHub personal access token
+   (https://github.com/settings/tokens), which will be used to create a GitHub
+   issue for each report. It requires `public_repo` scope. If omitted, no
+   issues will be created.
+ * `user` and `password` are a username/password pair which will be required to
+   access the bug report listings at `/api/listing`. If omitted, there will be
```
Comment (Kegsay) [inline README.md]: Probably worth mentioning this is HTTP basic auth, to remove all doubt.

**Comment by Kegsay** [comment]
LGTM - I imagine you'll say that some of my comments will be in another PR.

File: src/github.com/matrix-org/rageshake/submit.go (hunk 2 above)
Comment (richvdh) [inline src/github.com/matrix-org/rageshake/submit.go]: yup; more work to be done here.

File: src/github.com/matrix-org/rageshake/submit.go (hunk 3 above)
Comment (richvdh) [inline src/github.com/matrix-org/rageshake/submit.go]: > Just a placeholder for now - making it more useful will be a separate PR.

:-p

File: src/github.com/matrix-org/rageshake/submit.go (hunk 1 above)
Comment (richvdh) [inline src/github.com/matrix-org/rageshake/submit.go]: \<shrug>. Moved

File: README.md (hunk 4 above)
Comment (richvdh) [inline README.md]: done.
